frontend/src/pages/questions/UpdateQuestion.tsx

```diff
@@ -0,0 +1,223 @@
+} from '@chakra-ui/react';
+import {
+  AutoComplete,
+  AutoCompleteInput,
+  AutoCompleteItem,
+  AutoCompleteList,
+  AutoCompleteTag,
+} from '@choc-ui/chakra-autocomplete';
+
+import React, { useEffect, useState } from 'react';
+import QuestionsAPI from '../../api/questions/questions';
+import { useNavigate, useParams } from 'react-router-dom';
+import { type AxiosError } from 'axios';
+import { FaCheck } from 'react-icons/fa6';
+import { BiSolidBookAdd } from 'react-icons/bi';
+import IconWithText from '../../components/content/IconWithText';
+import ConfirmationDialog from '../../components/content/ConfirmationDialog';
+
+export const UpdateQuestion: React.FC = () => {
+  const { questionId } = useParams();
+  let questionIdString: string;
+  if (questionId !== undefined) {
+    questionIdString = questionId;
+  } else {
+    throw new Error('ID of question is undefined');
+  }
+  const [title, setTitle] = useState('');
+  const [questionDescription, setQuestionDescription] = useState('');
+  const [categories, setCategories] = useState<string[]>([]);
+  const [complexity, setComplexity] = useState('Easy');
+  const [linkToQuestion, setLinkToQuestion] = useState('');
+  const [allCategories, setAllCategories] = useState<string[]>([]);
+
+  const toast = useToast();
+
+  const linkPrefix = 'https://leetcode.com/problems/';
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    new QuestionsAPI()
+      .getCategories()
+      .then((categories) => {
+        setAllCategories(categories);
+      })
+      .catch(console.error);
+  }, []);
+
+  useEffect(() => {
+    new QuestionsAPI()
+      .getQuestion(questionIdString)
+      .then((questionData) => {
+        setTitle(questionData.title);
+        setQuestionDescription(questionData.questionDescription);
+        setCategories(questionData.categories);
+        setComplexity(questionData.complexity);
+        setLinkToQuestion(questionData.linkToQuestion.replace(linkPrefix, ''));
+      })
+      .catch((error) => {
+        console.error('Error fetching question data:', error);
+      });
+  }, []);
+
+  const handleSubmit: React.FormEventHandler<HTMLFormElement> = (e) => {
+    e.preventDefault();
+    new QuestionsAPI()
+      .updateQuestion(questionIdString, {
+        title,
+        questionDescription,
+        categories,
+        complexity,
+        linkToQuestion: linkPrefix + linkToQuestion,
+      })
+      .then(() => {
+        navigate('/');
+      })
+      .catch((err: AxiosError<{ errors: Array<{ msg: string }> }>) => {
+        const errors = err?.response?.data?.errors;
+        if (errors !== undefined) {
+          errors.map((error) =>
+            toast({
+              title: 'Question update failed.',
+              description: error.msg,
+              status: 'error',
+              duration: 9000,
+              isClosable: true,
+            }),
+          );
+        }
+      });
+  };
+
+  return (
+    <Card m={12} p={8}>
```

Comment:
significant code duplication from create - maybe we should refactor this to its own component but if too much work maybe future pr

Reply:
Done, refactored the code